### PR TITLE
Fixes #7258: Syntax error in specfile on rudder-agent

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://www.rudder-project.org/redmine/issues/7258

### CODE REVIEW

**Comment by VinceMacBuche** [comment]
PR updated

File: rudder-agent/SPECS/rudder-agent.spec [hunk 1]
```diff
@@ -297,16 +297,16 @@ fi
 
 # Test if compiler support hardening flags
 FILE=`mktemp`
```
Comment (Kegeruneku) [inline rudder-agent/SPECS/rudder-agent.spec]: I would instead do:
```FLAG_TEST_DIRECTORY=`mktemp -d`

echo "void main() {}" > "${FLAG_TEST_DIRECTORY}/test.c"
gcc -fPIE -pie -Wl,-zrelro -o "${FLAG_TEST_DIRECTORY}/test" "${FLAG_TEST_DIRECTORY}/test.c" 2>/dev/null

rm -rf ${FLAG_TEST_DIRECTORY}

File: rudder-agent/SPECS/rudder-agent.spec [hunk 2]
```diff
@@ -297,16 +297,16 @@ fi
...
 if [ $? -eq 0 ]
 then
   SECURE_CFLAGS="-fPIE -pie"
   SECURE_LDFLAGS="-z relro"
 fi
-rm -f "$${FILE}.o" "$${FILE}" ))
+rm -f "${FILE}.o" "${FILE}"
 
-./configure --build=%_target --prefix=%{rudderdir} --with-workdir=%{ruddervardir}/cfengine-community --enable-static=yes --enable-shared=no %{openssl_arg} %{lmdb_arg} CFLAGS="${SECURE_CFLAGS}" LDFLAGS="${SECURE_LDFLAGS}"
+./configure --build=%_target --prefix=%{rudderdir} --with-workdir=%{ruddervardir}/cfengine-community --enable-static=yes --enable-shared=no %{openssl_arg} %{lmdb_arg} CFLAGS="${SECURE_CFLAGS} ${CFLAGS}" LDFLAGS="${SECURE_LDFLAGS}"
```
Comment (Kegeruneku) [inline rudder-agent/SPECS/rudder-agent.spec]: I'd rather have CFLAGS at the beginning of the variable, so SECURE_CFLAGS would prevail over eventually conflicting options

File: rudder-agent/debian/rules [hunk 3]
```diff
@@ -26,18 +26,20 @@ configure-stamp: get-flags
 	# If there is no configure, bootstrap with autogen.sh first
 	cd SOURCES/cfengine-source && [ -x ./configure ] || NO_CONFIGURE=1 ./autogen.sh
 	# Prepare CFEngine 3.6.x build with defined path of LMDB
-	cd SOURCES/cfengine-source && ./configure --prefix=/opt/rudder --with-workdir=/var/rudder/cfengine-community --enable-static=yes --enable-shared=no --with-lmdb=$(CURDIR)/debian/tmp/opt/rudder CFLAGS="$(SECURE_CFLAGS)" LDFLAGS="$(SECURE_LDFLAGS)"
+	cd SOURCES/cfengine-source && ./configure --prefix=/opt/rudder --with-workdir=/var/rudder/cfengine-community --enable-static=yes --enable-shared=no --with-lmdb=$(CURDIR)/debian/tmp/opt/rudder CFLAGS="$(SECURE_CFLAGS) $(BUILD_CFLAGS)" LDFLAGS="$(SECURE_LDFLAGS) $(BUILD_LDFLAGS)"
```
Comment (Kegeruneku) [inline rudder-agent/debian/rules]: Do not forget to adjust this too when you have changed the SPECfile :) (and reverse the LDFLAGS too)

**Comment by VinceMacBuche** [comment]
PR updated

**Comment by VinceMacBuche** [comment]
PR updated

**Comment by peckpeck** [comment]
Good

**Pull request merged**